### PR TITLE
Fix crash in Microsoft.ML.Recommender with validation set

### PR DESCRIPTION
**Pull request by ankane**
This PR fixes a crash when calling `Fit` on the `MatrixFactorizationTrainer` class with a validation set. The crash is due to passing an incorrect length to LIBMF (it passes the length from the training set instead of the validation set).

### CODE REVIEW

**Review by michaelgsharp — APPROVED** [review]
LGTM. Thanks for the fix!

**Comment by ankane** [comment]
Thanks @michaelgsharp!
